hermes/serializers.py

```diff
@@ -138,59 +139,269 @@ class Meta(BaseTargetSerializer.Meta):
+            except:
+                try:
+                    dec_angle = Latitude(value)
+                except:
+                    raise serializers.ValidationError(_("Must be in a format astropy understands"))
+        return dec_angle.deg
 
-class PhotometrySerializer(serializers.Serializer):
+class CommonDataSerializer(serializers.Serializer):
     target_name = serializers.CharField(required=True)
-    ra = serializers.CharField(required=True)
-    dec = serializers.CharField(required=True)
-    date = serializers.CharField(required=True)
-    date_format = serializers.CharField(required=False)
+    date_obs = serializers.CharField(required=True)
     telescope = serializers.CharField(required=False, default='', allow_blank=True)
     instrument = serializers.CharField(required=False, default='', allow_blank=True)
-    band = serializers.CharField(required=True)
-    brightness = serializers.FloatField(required=True)
-    nondetection = serializers.BooleanField(required=False, default=False)
-    brightness_error = serializers.FloatField(required=False)
-    brightness_unit = serializers.ChoiceField(required=False, choices=["AB mag", "Vega mag", "mJy", "erg / s / cm² / Å"])
+
+    def validate_date_obs(self, value):
+        validate_date(value)
+        return value
 
     def validate(self, data):
         validated_data = super().validate(data)
-        validate_date(validated_data['date'], validated_data.get('date_format'))
         if not (validated_data.get('instrument') or validated_data.get('telescope')):
             error_msg = _("Must have at least one of telescope or instrument set")
-            raise serializers.ValidationError({'telescope': error_msg, 'instrument': error_msg})
+            raise serializers.ValidationError({'telescope': [error_msg], 'instrument': [error_msg]})
+        return validated_data
+
+
+class PhotometryDataSerializer(CommonDataSerializer):
+    bandpass = serializers.CharField(required=True)
+    brightness = serializers.FloatField(required=False)
+    brightness_error = serializers.FloatField(required=False)
+    brightness_unit = serializers.ChoiceField(required=False, default="AB mag", choices=["AB mag", "Vega mag", "mJy", "erg / s / cm² / Å"])
+    exposure_time = serializers.FloatField(required=False)
+    observer = serializers.CharField(required=False)
+    comments = serializers.CharField(required=False)
+    limiting_brightness = serializers.FloatField(required=False)
+    limiting_brightness_error = serializers.FloatField(required=False)
+    limiting_brightness_unit = serializers.ChoiceField(required=False, default="AB mag", choices=["AB mag", "Vega mag", "mJy", "erg / s / cm² / Å"])
+    group_associations = serializers.CharField(required=False)
+
+    def validate(self, data):
+        validated_data = super().validate(data)
+        if not (validated_data.get('brightness') or validated_data.get('limiting_brightness')):
```

Comment:
I'm not sure it makes sense to ever have both a brightness AND a limiting brightness.
@cmccully?

Reply:
I asked him this before and he said to allow both at the same time then at least...